### PR TITLE
Add `fractionalSecond` part type to DateTimeFormat().formatToParts

### LINKED ISSUE #48882: Intl.DateTimeFormat.formatToParts() does not include part of type `fractionalSecond`
# lib Update Request

## Configuration Check

My compilation *target* is `esnext` and my *lib* is `the default`.

## Missing / Incorrect Definition

`Intl.DateTimeFormat.prototype.formatToParts`

## Sample Code

`new Intl.DateTimeFormat().formatToParts().find((val) => val.type === 'fractionalSecond')`

## Documentation Link

Added in ECMAScript 2021: [https://402.ecma-international.org/#sec-formatdatetimepattern](https://402.ecma-international.org/#sec-formatdatetimepattern)

Also mentioned in MDN: [https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Intl/DateTimeFormat/formatToParts#description](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Intl/DateTimeFormat/formatToParts#description)

### PR DESCRIPTION
**Pull request by jordanmbell**
This change adds the `fractionalSecond` part type as a valid part
to be returned from Intl.DateTimeFormat().formatToParts().

Fixes #48882

### CODE REVIEW

**Comment by jordanmbell** [comment]
Let me know if I did something incorrectly. I couldn't tell if there was a different place I was supposed to add my change as this was not in in the original 2017 spec.

**Comment by sandersn** [comment]
Putting it into 2017 is a good simple way to do it even if it's not technically right. I'll check with the team to see if we can come up with another way to do it.

File: src/lib/es2017.intl.d.ts [hunk 1]
```diff
@@ -1,5 +1,5 @@
 declare namespace Intl {
-    type DateTimeFormatPartTypes = "day" | "dayPeriod" | "era" | "hour" | "literal" | "minute" | "month" | "second" | "timeZoneName" | "weekday" | "year";
+    type DateTimeFormatPartTypes = "day" | "dayPeriod" | "era" | "fractionalSecond" | "hour" | "literal" | "minute" | "month" | "second" | "timeZoneName" | "weekday" | "year";
```
Comment (sandersn) [inline src/lib/es2017.intl.d.ts:2]: suggestion from @weswigham:

```suggestion
    interface DateTimeFormatPartTypesRegistry {
        day: any
        dayPeriod: any
        // etc ...
    }
    type DateTimeFormatPartTypes = keyof DateTimeFormatPartTypesRegistry;
```

Then in es2021.intl.dts, add

```ts
    interface DateTimeFormatPartTypesRegistry {
       fractionalSecond: any
    }
```

**Review by sandersn — CHANGES_REQUESTED** [review]
Here's an idea for adding fractionSecond to ES2021 instead of ES2017.

**Pull request closed without merging**